### PR TITLE
Update Cilium extension to support Cilium 1.15

### PR DESCRIPTION
**Pull request by iljarotar**
Depends on #431

## Required Actions

```REQUIRED_ACTIONS
Clusters that use the Cilium CNI do not depend on MetalLB anymore and require new configuration keys in order to run with updated Cilium version `v0.15.4`. These networking configuration settings are `devices: ["lan+", "lo"]`, `bgpControlPlane.enabled: true` and `directRoutingDevice: lo`.
```

### CODE REVIEW

File: pkg/admission/mutator/defaulter.go [hunk 1]
```diff
@@ -96,6 +96,14 @@ func (d *defaulter) defaultNetworking(shoot *gardenv1beta1.Shoot) error {
...
 	}
 
+	if shoot.Spec.Networking.IPFamilies == nil || len(shoot.Spec.Networking.IPFamilies) == 0 {
+		ipFamiles := make([]gardenv1beta1.IPFamily, 0)
+		for _, f := range d.c.ipFamilies() {
+			ipFamiles = append(ipFamiles, gardenv1beta1.IPFamily(f))
+		}
+		shoot.Spec.Networking.IPFamilies = ipFamiles
+	}
+
```
Comment (robertvolkmann) [inline pkg/admission/mutator/defaulter.go:106]: Why do we need this?

File: pkg/admission/mutator/config.go [hunk 2]
```diff
@@ -60,23 +65,31 @@ func (c *config) ciliumKubeProxyEnabled() bool {
...
+	return c.bool("DEFAULTER_CILIUMBGPCONTROLPLANE", true)
 }
 
 func (c *config) ciliumIPv4NativeRoutingCIDREnabled() bool {
 	return c.bool("DEFAULTER_CILIUMIPV4NATIVEROUTINGCIDRENABLED", true)
 }
 
 func (c *config) ciliumLoadBalancingMode() ciliumextensionv1alpha1.LoadBalancingMode {
-	return ciliumextensionv1alpha1.LoadBalancingMode(c.string("DEFAULTER_CILIUMLOADBALANCINGMODE", string(ciliumextensionv1alpha1.DSR)))
+	return ciliumextensionv1alpha1.LoadBalancingMode(c.string("DEFAULTER_CILIUMLOADBALANCINGMODE", string(ciliumextensionv1alpha1.SNAT)))
```
Comment (robertvolkmann) [inline pkg/admission/mutator/config.go:92]: I still don't understand that is the desired value for metal-stack. Any idea?

**Review by robertvolkmann — CHANGES_REQUESTED** [review]

**Comment by robertvolkmann** [comment]
`metallb` is still by the internal helm chart `shoot-control-plane`.

**Review by robertvolkmann — APPROVED** [review]

**Comment by Gerrit91** [comment]
Needs rebase

**Pull request merged**